ambuda/xml.py

```diff
@@ -313,14 +313,21 @@ def parse_tei_header(blob: Optional[str]) -> dict[str, str]:
     }
 
 
-def transform_tei(blob: str) -> str:
-    """Transform XML for a TEI document."""
-    xml = ET.fromstring(blob)
-    return transform(xml, tei_xml)
-
-
 def transform_sak(blob: str) -> str:
     """Transform XML for the Shabdarthakaustubha."""
     xml = ET.fromstring(blob)
     # Reuse the Vacaspatyam xml config, since it's close enough.
     return transform(xml, vacaspatyam_xml)
+
+
+def transform_text_block(block_blob: str) -> Block:
+    """Transform XML for a TEI document."""
+    # FIXME: leaky abstraction. We should return just a string blob here and
```

Comment:
Not clear whether your alternative (returning a string) is actually better, won't that require an additional database call to get the XML ID?

Reply:
`block_blob` itself comes from `block.content`, so we'll have the block already without an extra query.